praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/service/PraktikumsstellenService.java

```diff
@@ -20,13 +31,69 @@ public class PraktikumsstellenService {
     }
 
     public AusbildungsPraktikumsstelleDTO saveAusbildungsPraktikumsstelle(final CreateAusbildungsPraktikumsstelleDTO createAusbildungsPraktikumsstelleDTO) {
+        AusbildungsPraktikumsstelle entityWithNormalDienststelle = praktikumsstellenMapper.toEntity(createAusbildungsPraktikumsstelleDTO,
+                meldezeitraumService.getCurrentMeldezeitraum());
+        entityWithNormalDienststelle.setDienststelle(normalizeDienststelle(entityWithNormalDienststelle.getDienststelle()));
         return praktikumsstellenMapper
                 .toDTO(ausbildungsPraktikumsstellenRepository
-                        .save(praktikumsstellenMapper.toEntity(createAusbildungsPraktikumsstelleDTO, meldezeitraumService.getCurrentMeldezeitraum())));
+                        .save(entityWithNormalDienststelle));
+    }
+
+    public TreeMap<String, List<PraktikumsstelleDTO>> getAllPraktiumsstellen() {
+        UUID lastMeldezeitraumID = meldezeitraumService.getMostRecentPassedMeldezeitraum().id();
+
+        Iterable<AusbildungsPraktikumsstelle> ausbildungsIterable = ausbildungsPraktikumsstellenRepository.findAllByMeldezeitraumID(lastMeldezeitraumID);
+        Iterable<StudiumsPraktikumsstelle> studiumsIterable = studiumsPraktikumsstellenRepository.findAllByMeldezeitraumID(lastMeldezeitraumID);
+
+        List<PraktikumsstelleDTO> ausbildungsListDTO = StreamSupport.stream(ausbildungsIterable.spliterator(), false)
+                .map(praktikumsstellenMapper::toDTO).collect(Collectors.toList());
+
+        List<PraktikumsstelleDTO> studiumsListDTO = StreamSupport.stream(studiumsIterable.spliterator(), false)
+                .map(praktikumsstellenMapper::toDTO).collect(Collectors.toList());
+
+        List<PraktikumsstelleDTO> combinedList = new ArrayList<>();
+        combinedList.addAll(ausbildungsListDTO);
+        combinedList.addAll(studiumsListDTO);
+        combinedList.sort(Comparator.comparing(PraktikumsstelleDTO::dienststelle));
+
+        TreeMap<String, List<PraktikumsstelleDTO>> groupedPraktikumsstellen = groupDienststellen(combinedList);
+
+        return groupedPraktikumsstellen;
+    }
+
+    private TreeMap<String, List<PraktikumsstelleDTO>> groupDienststellen(Iterable<PraktikumsstelleDTO> allPraktikumsstellen) {
+        TreeMap<String, List<PraktikumsstelleDTO>> abteilungsMap = new TreeMap<>();
+
+        for (PraktikumsstelleDTO praktikumsstelle : allPraktikumsstellen) {
+            String dienststelle = praktikumsstelle.dienststelle();
+            String hauptabteilung = getHauptabteilung(dienststelle);
+
+            abteilungsMap.computeIfAbsent(hauptabteilung, k -> new ArrayList<>()).add(praktikumsstelle);
+        }
+
+        return abteilungsMap;
+    }
+
+    private String getHauptabteilung(String dienststelle) {
+        int index = -1;
+        for (int i = 0; i < dienststelle.length(); i++) {
+            if (Character.isDigit(dienststelle.charAt(i))) {
+                index = i;
+                break;
```

Comment:
I think a return would be better to understand what happens and then return dienststelle by default if the loop did not trigger the return